lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -23,16 +23,33 @@ type DialogType =
   | null;
 
 export type ContentSelectorProps = {
+  initialContent?: Content;
+
   onError: (ei: ErrorInfo) => void;
   onSelectContent: (c: Content) => void;
+
   /** Used for testing */
   defaultActiveDialog?: DialogType;
 };
 
+function extractContentTypeAndValue(content: Content): [DialogType, string] {
+  if (content.type === 'url' && content.url.match(/^https?:/i)) {
+    return ['url', content.url];
+  } else if (content.type === 'url' && content.url.startsWith('jstor:')) {
+    return ['jstor', content.url.slice('jstor://'.length)];
+  } else {
+    // Other content types are not currently handled, which means that the user
+    // will need to select content starting from a blank state, as opposed to
+    // being able to adjust the selection.
+    return [null, ''];
+  }
+}
```

Comment:
Maybe we can make this function accept `undefined | Content`, that way we don't have to handle the "fallback" value later in the `useMemo` where this function is called.

It would require null checks thought, which are not super beatuful, so I leave it up to you:

```suggestion
function extractContentTypeAndValue(content?: Content): [DialogType, string] {
  if (content?.type === 'url' && content?.url.match(/^https?:/i)) {
    return ['url', content.url];
  } else if (content?.type === 'url' && content?.url.startsWith('jstor:')) {
    return ['jstor', content.url.slice('jstor://'.length)];
  }

  // Other content types are not currently handled, which means that the user
  // will need to select content starting from a blank state, as opposed to
  // being able to adjust the selection.
  return [null, ''];
}
```